src/AbstractMappingTransformer.php

```diff
@@ -49,7 +51,13 @@ protected function transformFromMapping(array $mapping, array $data)
 
                 // If we didn't get value but field was required, throw OptimusException
                 if (isset($node['required']) && true === $node['required'] && null === $nodeValue) {
-                    throw new OptimusException('Field ' . $node['from'] . ' required.');
+                    $message = 'Field ' . $node['from'] . ' required.';
+                    if (!$returnAllErrors) {
+                        throw new OptimusException($message);
+                    } else {
+                        $errors[] = $message;
+                        continue;
```

Comment:
This `if ... else` could be factorize via a private method `$this->throwException($message)` ?

Reply:
Not sure, i need to go to next iteration only on else case. In a private method i can't do that.